cmd/feed-ingress/main.go

```diff
@@ -375,6 +395,11 @@ func createIngressUpdaters() ([]controller.Updater, error) {
 		updaters = append(updaters, gorbUpdater)
 
 	case "merlin":
+		if internalHostname == "" || externalHostname == "" {
+			return updaters, errors.New("Unable to update ingress for Merlin frontends. Requires both " +
+				"'internal-hostname' and 'external-hostname' flags.")
+		}
```

Comment:
Why do we need both internal/external VIPs/hostnames? In m25 now we only have the internal one. I don't see why this is required either, if neither are provided just don't update the status.

Also, we should be able to query these values directly from merlin (list the virtual services and find their VIPs). So no need to require them as flags.

Reply:
Also, the names "internal" "external" should match the merlin flag names, which are "http-service" and "https-service". So what you want here is "httpServiceVIP", and "httpsServiceVIP". But like I said, we should just query these from merlin at startup.

Reply:
Ok, I understand a bit better now, it is trying to match the AWS approach. But for M25, we don't have the concept of external or internal ingress like we do in AWS (where there is a separate external ELB and internal ELB). In M25 all ingresses are accessible through both the internal VIPs, as well as the external VIPs assigned to the palo alto. The internal setup could also have multiple VIPs, one for each service (http and https). I think we can derive the internal VIPs from merlin directly, but we can't do that for the "external" VIP. This is a tricky one. It seems like we need to provide that external VIP (or hostname, if that's needed) only, and then we need to use the frontend-scheme annotation to indicate the value to use so external-dns can assign the correct value. Need to think about it.

Reply:
Does external-dns support creating explicit separate DNS entries for an ingress, via annotation or something? For an ingress using merlin, I think the load balancer status should contain the ingress VIPs. Any other hostname/vip is custom and depends entirely on the local setup, so I'm thinking this should be handled with a different mechanism. Although using the annotation to override that VIP is one approach (but will mean we likely need separate ingress for internal and external which isn't really necessary like it is with AWS).

Reply:
I was trying to use the same logic that was used in `feed-dns` (where the VIP's are defined as flags). Surely the `http` and `https` services run on the same IP(s) so surely we can just use the `merlinInstanceIP` as the only 'internal' ingress point? Once 'external' ingress has been added we can update the merlin status updater accordingly.

Reply:
I have added a new `merlinInternetFacingVIP` flag which for now will just be metadata for the frontend ip address (will only be used for setting DNS and general information about the status). Going forward, we will probably want to grab both this value and `merlinVIP` directly from Merlin.